bin/picom-trans

```diff
@@ -102,54 +106,56 @@ parse_args()
     i=$(( i + 1 ))
   done
 
-
-  while test $# -gt 0; do
-    # 15 +15 -15 are opacity values
-    # filter them out here, otherwise getopts will complain
-    if echo "$1" | grep -qE '^[+-]?[[:digit:]]+%?$'; then
-      target_opacity=$1
-      shift
-      continue
+  # Iterate over every argument and check if it is an opacity without the -o
+  # option in the previous argument. If so, then prepend the -o option.
+  # e.g. Turn this -
+  #     picom-trans -c +10 -s
+  # into this -
+  #     picom-trans -c -o +10 -s
+  i=1
+  while [ $i -le $# ]
+  do
+    if echo "$1" | grep -qE '^[+-]?[[:digit:]]+%?$' && \
+      ! eval "echo \${$#}" | grep -q '^-[hdtrgsnwc]*o$'
+    # NOTE: eval "echo \${$#}" means 'echo the last argument'
+    # NOTE: The letters inside the square brackets (ie. hdtrgsnwc) are
+    #       the same as those in the getopts argument, but without the 'o'
+    then
+      OPACITY=$1
+      shift && set -- "-o" "$OPACITY" "$@"
```

Comment:
We are checking the `OPACITY` value twice (once to add `-o` and then again after adding `-o`). But the current approach is probably more readable.

Reply:
I think readability is more important.

What do you think?

Reply:
e48cf95